### PR TITLE
fix: disable save button when workspace name is empty 

### LINKED ISSUE #2260: Disable Save Button if workspace name is empty
### Preliminary Checks

- [X] I have reviewed the documentation: https://unkey.com/docs

- [X] I have searched for existing issues: https://github.com/unkeyed/unkey/issues

- [X] This issue is not a question, general help request, or anything other than a bug report directly related to Unkey. Please ask questions in our Discord community: https://unkey.com/discord.

### Reproduction / Replay Link (Optional)

_No response_

### Issue Summary

When i submit the button with empty workspace name its submitting 

### Steps to Reproduce

1. In settings > general 
2. workspace name 

### Expected behavior

Disable the save button if the name is empty 

### Other information

_No response_

### Screenshots

[screen-capture (25).webm](https://github.com/user-attachments/assets/3051c275-2547-4788-af6c-5ad63fd1535e)

### Version info

```markdown
- OS:Windows 
- Node:v20.17.0
- npm:10.8.2
```

### PR DESCRIPTION
**Pull request by RajuGangitla**
## What does this PR do?

This pr will a add validation when workspace name is empty 

Fixes #2260 
[screen-capture (21).webm](https://github.com/user-attachments/assets/8cd837d3-f03d-4f94-972e-8140a8158ac0)

## Summary by CodeRabbit

- **New Features**
	- Enhanced role and permission management on the API key detail page.
	- Introduced `PermissionTree` component for improved display of roles and permissions.
	- Updated workspace name validation with new regex and minimum length requirements.
	- Added new API endpoints: `createApi`, `listKeys`, and `deleteApi`.

- **Bug Fixes**
	- Improved font rendering behavior across various applications.

- **Documentation**
	- Updated TypeScript definitions for clarity and added new API endpoints.

- **Chores**
	- Enhanced internal logic for managing permissions while maintaining existing verification data handling.

### CODE REVIEW

**Comment by chronark** [comment]
There seems to be a lot going on in this PR that is unrelated to fixing the button, could you check?

**Comment by RajuGangitla** [comment]
@chronark it is because of my other pr which was not merged  #2238 
and im getting that same code here also 
can you merge that pr first

**Comment by chronark** [comment]
looks like a lot of formatting issues, can you run `pnpm fmt` on this?

**Comment by RajuGangitla** [comment]
done

**Comment by chronark** [comment]
still blocked by #2238, unless you clean it up

**Comment by RajuGangitla** [comment]
@chronark i resolved #2238 after merging that issue i will resolve this one

File: apps/dashboard/app/(app)/settings/general/update-workspace-name.tsx
Comment (chronark) [inline apps/dashboard/app/(app)/settings/general/update-workspace-name.tsx:95]: put it all in here or all in a separate variable, but please don't do it half and half

**Review by chronark — CHANGES_REQUESTED** [review]

File: apps/dashboard/app/layout.tsx
Comment (RajuGangitla) [reply apps/dashboard/app/layout.tsx:18]: It's because of that previous pr 

I will resolve all issues

File: apps/api/src/pkg/keys/service.ts
Comment (chronark) [reply apps/api/src/pkg/keys/service.ts:271]: no that's a table relation, not column selection

**Review by chronark — APPROVED** [review]

**Review by chronark — APPROVED** [review]

**Pull request merged**
